PyPDF2/_page.py

```diff
@@ -1215,56 +1271,101 @@ def process_operation(operator: bytes, operands: List) -> None:
                     float(operands[5]),
                 ]
             elif operator == b"T*":
+                check_crlf_space = True
                 tm_matrix[5] -= TL
+
             elif operator == b"Tj":
-                t: str = ""
-                tt: bytes = (
-                    encode_pdfdocencoding(operands[0])
-                    if isinstance(operands[0], str)
-                    else operands[0]
-                )
-                if isinstance(cmap[0], str):
-                    try:
-                        t = tt.decode(cmap[0], "surrogatepass")  # apply str encoding
-                    except Exception:  # the data does not match the expectation, we use the alternative ; text extraction may not be good
-                        t = tt.decode(
-                            "utf-16-be" if cmap[0] == "charmap" else "charmap",
-                            "surrogatepass",
-                        )  # apply str encoding
-                else:  # apply dict encoding
-                    t = "".join(
-                        [
-                            cmap[0][x] if x in cmap[0] else bytes((x,)).decode()
-                            for x in tt
-                        ]
+                check_crlf_space = True
+                if isinstance(operands[0], str):
+                    text += operands[0]
+                else:
+                    t: str = ""
+                    tt: bytes = (
+                        encode_pdfdocencoding(operands[0])
+                        if isinstance(operands[0], str)
+                        else operands[0]
                     )
```

Comment:
Good catch. Do you want to submit a corresponding PR?

Reply:
@biredel 
I'm confused : 
you are looking at an obsolete branch (PyPDF2) instead of pypdf.
The code seems to be this one
https://github.com/py-pdf/pypdf/blob/a435eaaa08c71e3f66320edd06be24637ef32986/pypdf/_text_extraction/__init__.py#L225C18-L234C18

Codecov indicates some test coverage.